### PR TITLE
Add Ripple provider

### PR DESCRIPTION
**Pull request by shahbaz17**

### CODE REVIEW

**Review by develoQ — COMMENTED** [review]
`XRPL` and `Ripple` are not identical.

It is appropriate to use `XRPL` here.
(`XRP` for currency).

XRP Ledger website is as follows
https://xrpl.org

File: docs/connect-blockchain/connect-blockchain.mdx [hunk 1]
```diff
@@ -80,6 +81,41 @@ const userAccounts = await solanaWallet.requestAccounts();
 console.log(userAccounts);
 ```
 
+</TabItem>
+<TabItem value="ripple">
```
Comment (develoQ) [inline docs/connect-blockchain/connect-blockchain.mdx:85]:
```suggestion
<TabItem value="XRPL">
```

File: docs/connect-blockchain/connect-blockchain.mdx [hunk 2]
```diff
@@ -80,6 +81,41 @@ const userAccounts = await solanaWallet.requestAccounts();
...
+const adapter = new OpenloginAdapter({
+adapterSettings: {
+  provider: xrplProvider, // <-- Injecting the Ripple provider
+},
+});
+web3AuthInstance.configureAdapter(adapter);
+
+await web3auth.initModal();
+
+const rippleProvider = await web3auth.connect();
```
Comment (develoQ) [inline docs/connect-blockchain/connect-blockchain.mdx:116]:
```suggestion
const xrplProvider = await web3auth.connect();
```

File: docs/connect-blockchain/connect-blockchain.mdx [hunk 3]
```diff
@@ -134,7 +170,8 @@ authentication. This provider can be used to interact with wallet or connected c
 can follow our API Reference to know more about how to use these providers with their compatible adapters:
 
 - [EIP1193 (EVM) Providers](/sdk/helper-sdks/providers/evm)
-- [Solana Providers](/sdk/pnp/web/providers/solana)
+- [Solana Providers](/sdk/helper-sdks/providers/solana)
+- [Ripple Providers](/sdk/helper-sdks/providers/ripple)
```
Comment (develoQ) [inline docs/connect-blockchain/connect-blockchain.mdx:174]:
```suggestion
- [XRPL Providers](/sdk/helper-sdks/providers/xrpl)
```

File: docs/connect-blockchain/ripple.mdx [hunk 4]
```diff
@@ -0,0 +1,189 @@
+---
+title: Integrate Web3Auth with the Ripple Blockchain
+sidebar_label: Ripple
+image: "content-hub/guides/banners/ripple.png"
```
Comment (develoQ) [inline docs/connect-blockchain/ripple.mdx:4]:
```suggestion
title: Integrate Web3Auth with the XRPL Blockchain
sidebar_label: XRPL
image: "content-hub/guides/banners/xrpl.png"
```

File: docs/connect-blockchain/ripple.mdx [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+displayed_sidebar: docs
+keywords: [ripple, web3auth, authentication, blockchain]
+description: "Integrate Web3Auth with the Ripple Blockchain | Documentation - Web3Auth"
```
Comment (develoQ) [inline docs/connect-blockchain/ripple.mdx:7]:
```suggestion
description: "Integrate Web3Auth with the XRPL Blockchain | Documentation - Web3Auth"
```

File: docs/connect-blockchain/ripple.mdx [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+---
+
+import GetUserInfoSnippet from "@site/src/common/docs/_get-userinfo.mdx";
+
+---
+
+The Web3Auth Web SDK offers a standard provider for accessing the user's private key on non-EVM chains like [Ripple](https://ripple.com/). To enable a
```
Comment (develoQ) [inline docs/connect-blockchain/ripple.mdx:14]:
```suggestion
The Web3Auth Web SDK offers a standard provider for accessing the user's private key on non-EVM chains like [XRPL](https://xrpl.org/). To enable a
```

File: docs/connect-blockchain/ripple.mdx [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+
+const xrplProvider = new XrplPrivateKeyProvider({
+  config: {
+    chainConfig: getXRPLChainConfig("testnet"), // devnet, testnet, mainnet
+  },
+});
+
+const adapter = new OpenloginAdapter({
+  adapterSettings: {
+    provider: xrplProvider, // <-- Injecting the Ripple provider
```
Comment (develoQ) [inline docs/connect-blockchain/ripple.mdx:52]:
```suggestion
    provider: xrplProvider, // <-- Injecting the XRPL provider
```

File: docs/connect-blockchain/ripple.mdx [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+  },
+});
+web3AuthInstance.configureAdapter(adapter);
+
+await web3auth.initModal();
+
+const rippleProvider = await web3auth.connect(); // web3auth.provider
```
Comment (develoQ) [inline docs/connect-blockchain/ripple.mdx:59]:
```suggestion
const xrplProvider = await web3auth.connect(); // web3auth.provider
```

File: docs/connect-blockchain/ripple.mdx [hunk 9]
```diff
@@ -0,0 +1,189 @@
...
+
+```js
+try {
+  // rippleProvider is from above
+  const accounts = await rippleProvider.request<string[]>({
+    method: "xrpl_getAccounts",
+  });
+
+  if (accounts) {
+    const accInfo = (await rippleProvider.request({
```
Comment (develoQ) [inline docs/connect-blockchain/ripple.mdx:80]:
```suggestion
    const accInfo = (await xrplProvider.request({
```

File: docs/connect-blockchain/ripple.mdx [hunk 10]
```diff
@@ -0,0 +1,189 @@
...
+
+  if (accounts && accounts.length > 0) {
+    const tx: Payment = {
+      TransactionType: "Payment",
+      Account: accounts[0] as string,
+      Amount: xrpToDrops(2),
+      Destination: "rJAHHPYmy4g3h7kzfj2Mzm2nHwpKuVdEvX", // Destination address
+    };
+    const txSign = await provider.request({
+      method: "ripple_signTransaction",
```
Comment (develoQ) [inline docs/connect-blockchain/ripple.mdx:122]:
```suggestion
      method: "xrpl_signTransaction",
```

File: docs/connect-blockchain/ripple.mdx [hunk 11]
```diff
@@ -0,0 +1,189 @@
...
+
+  if (accounts && accounts.length > 0) {
+    const tx: Payment = {
+      TransactionType: "Payment",
+      Account: accounts[0] as string,
+      Amount: xrpToDrops(2),
+      Destination: "rJAHHPYmy4g3h7kzfj2Mzm2nHwpKuVdEvX",
+    };
+    const txSign = await provider.request({
+      method: "ripple_submitTransaction",
```
Comment (develoQ) [inline docs/connect-blockchain/ripple.mdx:154]:
```suggestion
      method: "xrpl_submitTransaction",
```

File: docs/connect-blockchain/ripple.mdx [hunk 12]
```diff
@@ -0,0 +1,189 @@
+---
```
Comment (develoQ) [inline docs/connect-blockchain/ripple.mdx:1]: The file name should be `xrpl.mdx`.

File: docs/sdk/helper-sdks/providers/ripple.mdx [hunk 13]
```diff
@@ -0,0 +1,210 @@
+---
```
Comment (develoQ) [inline docs/sdk/helper-sdks/providers/ripple.mdx:1]: The file name should be `xrpl.mdx`.

File: docs/sdk/helper-sdks/providers/ripple.mdx [hunk 14]
```diff
@@ -0,0 +1,210 @@
...
+title: Ripple Provider
+displayed_sidebar: sdk
+description: "Ripple Provider | Documentation - Web3Auth"
```
Comment (develoQ) [inline docs/sdk/helper-sdks/providers/ripple.mdx:4]:
```suggestion
title: XRPL Provider
displayed_sidebar: sdk
description: "XRPL Provider | Documentation - Web3Auth"
```

File: docs/sdk/helper-sdks/providers/ripple.mdx [hunk 15]
```diff
@@ -0,0 +1,210 @@
...
+## Usage
+
+---
+
+After configuring the provider, you may utilize various functions from the `@web3auth/xrpl-provider` library for tasks such as obtaining the user's
+account, executing transactions, and signing messages. Below are a few sample use cases to assist you in getting started:
+
+### Get User Account and Balance
+
+#### `ripple_getAccounts`
```
Comment (develoQ) [inline docs/sdk/helper-sdks/providers/ripple.mdx:59]:
```suggestion
#### `xrpl_getAccounts`
```

File: docs/sdk/helper-sdks/providers/ripple.mdx [hunk 16]
```diff
@@ -0,0 +1,210 @@
...
+The purpose of this function is to retrieve the details of the connected account.
+
+#### Example
+
+```ts
+try {
+  // rippleProvider is from above
+  const accounts = await rippleProvider.request<string[]>({
+    method: "xrpl_getAccounts",
+  });
```
Comment (develoQ) [inline docs/sdk/helper-sdks/providers/ripple.mdx:70]:
```suggestion
  // xrplProvider is from above
  const accounts = await xrplProvider.request<string[]>({
    method: "xrpl_getAccounts",
  });
```

File: docs/sdk/helper-sdks/providers/ripple.mdx [hunk 17]
```diff
@@ -0,0 +1,210 @@
...
+
+  if (accounts) {
+    const accInfo = (await rippleProvider.request({
```
Comment (develoQ) [inline docs/sdk/helper-sdks/providers/ripple.mdx:73]:
```suggestion
    const accInfo = (await xrplProvider.request({
```

File: docs/sdk/helper-sdks/providers/ripple.mdx [hunk 18]
```diff
@@ -0,0 +1,210 @@
...
+    console.log("No accounts found, please report this issue.");
+  }
+} catch (error) {
+  console.error("Error", error);
+}
+```
+
+### Sign a Transaction
+
+#### `ripple_signTransaction`
```
Comment (develoQ) [inline docs/sdk/helper-sdks/providers/ripple.mdx:99]:
```suggestion
#### `xrpl_signTransaction`
```

File: docs/sdk/helper-sdks/providers/ripple.mdx [hunk 19]
```diff
@@ -0,0 +1,210 @@
...
+
+  if (accounts && accounts.length > 0) {
+    const tx: Payment = {
+      TransactionType: "Payment",
+      Account: accounts[0] as string,
+      Amount: xrpToDrops(2),
+      Destination: "rJAHHPYmy4g3h7kzfj2Mzm2nHwpKuVdEvX", // Destination address
+    };
+    const txSign = await provider.request({
+      method: "ripple_signTransaction",
```
Comment (develoQ) [inline docs/sdk/helper-sdks/providers/ripple.mdx:119]:
```suggestion
      method: "xrpl_signTransaction",
```

File: docs/sdk/helper-sdks/providers/ripple.mdx [hunk 20]
```diff
@@ -0,0 +1,210 @@
...
+    console.log("failed to fetch accounts");
+  }
+} catch (error) {
+  console.log("error", error);
+}
+```
+
+### Sign and Send a Transaction
+
+#### `ripple_submitTransaction`
```
Comment (develoQ) [inline docs/sdk/helper-sdks/providers/ripple.mdx:135]:
```suggestion
#### `xrpl_submitTransaction`
```

File: docs/sdk/helper-sdks/providers/ripple.mdx [hunk 21]
```diff
@@ -0,0 +1,210 @@
...
+
+  if (accounts && accounts.length > 0) {
+    const tx: Payment = {
+      TransactionType: "Payment",
+      Account: accounts[0] as string,
+      Amount: xrpToDrops(2),
+      Destination: "rJAHHPYmy4g3h7kzfj2Mzm2nHwpKuVdEvX",
+    };
+    const txSign = await provider.request({
+      method: "ripple_submitTransaction",
```
Comment (develoQ) [inline docs/sdk/helper-sdks/providers/ripple.mdx:156]:
```suggestion
      method: "xrpl_submitTransaction",
```

File: docs/sdk/helper-sdks/providers/ripple.mdx [hunk 22]
```diff
@@ -0,0 +1,210 @@
...
+    console.log("failed to fetch accounts");
+  }
+} catch (error) {
+  console.log("error", error);
+}
+```
+
+### Sign a Message
+
+#### `ripple_signMessage`
```
Comment (develoQ) [inline docs/sdk/helper-sdks/providers/ripple.mdx:172]:
```suggestion
#### `xrpl_signMessage`
```

File: sidebars.js [hunk 23]
```diff
@@ -196,6 +196,7 @@ module.exports = {
           ],
         },
         "connect-blockchain/solana",
+        "connect-blockchain/ripple",
```
Comment (develoQ) [inline sidebars.js:199]:
```suggestion
        "connect-blockchain/xrpl",
```

File: sidebars.js [hunk 24]
```diff
@@ -600,6 +601,7 @@ module.exports = {
       items: [
         "sdk/helper-sdks/providers/evm",
         "sdk/helper-sdks/providers/solana",
+        "sdk/helper-sdks/providers/ripple",
```
Comment (develoQ) [inline sidebars.js:604]:
```suggestion
        "sdk/helper-sdks/providers/xrpl",
```

File: src/common/guides/_interacting-with-blockchain.mdx [hunk 25]
```diff
@@ -10,7 +10,8 @@ for EVM and Solana blockchains and for others, you can get the private key in th
 provider, Web3Auth Base provides a `SafeEventEmitterProvider` from which you can create your own provider.
 
 - [Ethereum Provider](/sdk/helper-sdks/providers/evm) gives you the capability of making RPC calls to the EVM compatible blockchains.
-- [Solana Provider](/sdk/pnp/web/providers/solana) gives you the capability of making RPC calls to the Solana blockchain.
+- [Solana Provider](/sdk/helper-sdks/providers/solana) gives you the capability of making RPC calls to the Solana blockchain.
+- [Ripple Provider](/sdk/helper-sdks/providers/ripple) gives you the capability of making RPC calls to the Ripple blockchain.
```
Comment (develoQ) [inline src/common/guides/_interacting-with-blockchain.mdx:14]:
```suggestion
- [XRPL Provider](/sdk/helper-sdks/providers/xrpl) gives you the capability of making RPC calls to the XRPL blockchain.
```

File: static/content-hub/guides/banners/ripple.png
Comment (develoQ) [inline static/content-hub/guides/banners/ripple.png:1]: The file name should be `xrpl.png`.

File: docs/sdk/helper-sdks/providers/ripple.mdx [hunk 26]
```diff
@@ -0,0 +1,210 @@
...
+const adapter = new OpenloginAdapter({
+  adapterSettings: {
+    provider: xrplProvider, // <-- Injecting the Ripple provider
+  },
+});
+web3AuthInstance.configureAdapter(adapter);
+
+await web3auth.initModal();
+
+const rippleProvider = await web3auth.connect();
```
Comment (develoQ) [inline docs/sdk/helper-sdks/providers/ripple.mdx:47]:
```suggestion
const xrplProvider = await web3auth.connect();
```

**Comment by shahbaz17** [comment]
Hey @develoQ 

All reference of ripple is changes to xrpl

File: docs/sdk/helper-sdks/providers/xrpl.mdx [hunk 27]
```diff
@@ -0,0 +1,210 @@
...
+## Usage
+
+---
+
+After configuring the provider, you may utilize various functions from the `@web3auth/xrpl-provider` library for tasks such as obtaining the user's
+account, executing transactions, and signing messages. Below are a few sample use cases to assist you in getting started:
+
+### Get User Account and Balance
+
+#### `ripple_getAccounts`
```
Comment (develoQ) [inline docs/sdk/helper-sdks/providers/xrpl.mdx:59]:
```suggestion
#### `xrpl_getAccounts`
```

File: docs/sdk/helper-sdks/providers/xrpl.mdx [hunk 28]
```diff
@@ -0,0 +1,210 @@
...
+
+  if (accounts && accounts.length > 0) {
+    const tx: Payment = {
+      TransactionType: "Payment",
+      Account: accounts[0] as string,
+      Amount: xrpToDrops(2),
+      Destination: "rJAHHPYmy4g3h7kzfj2Mzm2nHwpKuVdEvX", // Destination address
+    };
+    const txSign = await provider.request({
+      method: "ripple_signTransaction",
```
Comment (develoQ) [inline docs/sdk/helper-sdks/providers/xrpl.mdx:119]:
```suggestion
      method: "xrpl_signTransaction",
```

File: docs/sdk/helper-sdks/providers/xrpl.mdx [hunk 29]
```diff
@@ -0,0 +1,210 @@
...
+
+  if (accounts && accounts.length > 0) {
+    const tx: Payment = {
+      TransactionType: "Payment",
+      Account: accounts[0] as string,
+      Amount: xrpToDrops(2),
+      Destination: "rJAHHPYmy4g3h7kzfj2Mzm2nHwpKuVdEvX",
+    };
+    const txSign = await provider.request({
+      method: "ripple_submitTransaction",
```
Comment (develoQ) [inline docs/sdk/helper-sdks/providers/xrpl.mdx:156]:
```suggestion
      method: "xrpl_submitTransaction",
```

**Review by develoQ — CHANGES_REQUESTED** [review]
I see that there were a few items left, so I will request them!

Thank you for your kind!

**Review by develoQ — APPROVED** [review]
Thank you for your time!

**Pull request merged**